File: 1468-check-if-n-and-its-double-exist/1468-check-if-n-and-its-double-exist.cpp
class Solution {
public:
    bool checkIfExist(vector<int>& arr) {
        unordered_set<int> st;
        for(auto it : arr) {
            if(st.find(2*it) != st.end()) return true;
            if(it%2 == 0 && st.find(it/2) != st.end()) return true;
            st.insert(it);
        }

        return false;
    }
};

File: 1468-check-if-n-and-its-double-exist/1468-check-if-n-and-its-double-exist.py
class Solution(object):
    def checkIfExist(self, arr):
        st = set()
        for it in arr:
            if it*2 in st:
                return True
            if it%2 == 0 and it//2 in st:
                return True
            st.add(it)
            
        return False
        